features/dd-sdk-android-flags/src/main/kotlin/com/datadog/android/flags/featureflags/FlagsClient.kt

```diff
@@ -10,114 +10,175 @@ import com.datadog.android.Datadog
 
-    // region Internal
+    /**
+     * Resolve a Int flag value.
+     * @param flagKey The name of the key to query.
+     * @param defaultValue The value to return if the key cannot be retrieved.
+     * @return The value of the flag, or the default value if it cannot be retrieved.
+     */
+    fun resolveIntValue(flagKey: String, defaultValue: Int): Int
 
     /**
-     * Register a [FlagsProvider] with an [SdkCore] to back the behaviour of the [get] function.
-     *
-     * Registration is a one-time operation. Once a provider has been registered, all attempts at re-registering
-     * will return `false`.
-     *
-     * Every application intending to use the global FlagsClient is responsible for registering it once
-     * during its initialization.
+     * Resolve a Structure flag value.
+     * @param flagKey The name of the key to query.
+     * @param defaultValue The value to return if the key cannot be retrieved.
+     * @return The value of the flag, or the default value if it cannot be retrieved.
+     */
+    fun resolveStructureValue(flagKey: String, defaultValue: JSONObject): JSONObject
+
+    /**
+     * Companion object providing static access to [FlagsClient] instances.
      *
-     * @param provider the provider to use as global provider.
-     * @param sdkCore the instance to register the given provider with. If not provided, default
-     * instance will be used.
-     * @return `true` if the provider was registered as a result of this call, `false` otherwise.
+     * This companion manages the registration and retrieval of [FlagsClient] instances
+     * per SDK core, ensuring thread-safe access and proper lifecycle management.
      */
-    internal fun registerIfAbsent(provider: FlagsProvider, sdkCore: SdkCore = Datadog.getInstance()): Boolean =
-        synchronized(registeredProviders) {
-            if (registeredProviders.containsKey(sdkCore)) {
-                (sdkCore as FeatureSdkCore).internalLogger.log(
-                    InternalLogger.Level.WARN,
-                    InternalLogger.Target.USER,
-                    { "A FlagsProvider has already been registered for this SDK instance" }
-                )
-                false
+    companion object {
+        private val registeredClients: MutableMap<SdkCore, FlagsClient> = mutableMapOf()
+
+        /**
+         * Returns the [FlagsClient] instance for the given SDK core.
+         *
+         * This method is thread-safe and will return the same client instance for the same SDK core
+         * across multiple calls. If no client has been registered for the given SDK core, a no-op
+         * implementation will be returned instead.
+         *
+         * @param sdkCore the [SdkCore] instance to retrieve the client for. If not provided,
+         * the default Datadog SDK instance will be used.
+         * @return the [FlagsClient] associated with the given SDK core, or a [NoOpFlagsClient]
```

Comment:
We shouldn't reference `NoOpFlagsClient` as a code symbol from the public API, because it is `internal`, so not accessible to the customer anyway. I suggest to replace a wording with `or a no-op client`, like [here](https://github.com/DataDog/dd-sdk-android/blob/6de04ce550bd7364b62e51dbeccc4ac8afac90b7/features/dd-sdk-android-rum/src/main/kotlin/com/datadog/android/rum/GlobalRumMonitor.kt#L52).